common/src/main/java/com/microsoft/identity/common/internal/ui/webview/challengehandlers/CertBasedAuthFactory.java

```diff
@@ -73,13 +77,24 @@ public void onException(@NonNull final Exception exception) {
      * @param callback logic to run after a ICertBasedAuthChallengeHandler is chosen.
      */
     public void createCertBasedAuthChallengeHandler(@NonNull final CertBasedAuthChallengeHandlerCallback callback) {
+        final CertBasedAuthTelemetryHelper telemetryHelper = new CertBasedAuthTelemetryHelper();
+        telemetryHelper.setUserChoice(NON_APPLICABLE);
```

Comment:
Wouldn't this be considered `on-device`?

Reply:
I don't think so, since the user will still be on the UserChoice dialog and have not signaled that they want to use on-device yet. Unless you're referring to the if-block starting on line 83, which I don't consider as a user choice since this means they're just not able to use smartcard CBA at all (so on-device is the default).